### PR TITLE
Prevent log file from being overwritten, redirect STDOUT even when not run as daemon, log timestamp

### PR DESCRIPTION
**Pull request by amalek215**

### CODE REVIEW

File: bin/pgaudit_analyze [hunk 1]
```diff
@@ -670,7 +680,7 @@ while(!$bDone)
             $strLogFile = $strNextLogFile;
             undef($strNextLogFile);
 
-            syswrite($hLog, "reading ${strLogFile}\n");
+            syswrite($hLog, "${\getTimestamp()}: reading ${strLogFile}\n");
```
Comment (dwsteele) [inline bin/pgaudit_analyze:683]: This is a good idea, but we should also do it anywhere else we write to the log file.

File: bin/pgaudit_analyze [hunk 2]
```diff
@@ -630,9 +636,13 @@ my $bDone = false;
...
 # Open log file
-open(my $hLog, '>', $strLogOutFile)
+open(my $hLog, '>>', $strLogOutFile)
     or confess "unable to open pgAudit Analyze log file $strLogOutFile: $!";
 
+# always redirect stdout to logfile
+open (STDOUT, '>&', $hLog)
+    or confess "Couldn't close standard output and redirect to logfile: $!";
+
 # Daemonize the process
```
Comment (dwsteele) [inline bin/pgaudit_analyze:646]: Why would you like to make this change?

**Review by dwsteele — CHANGES_REQUESTED** [review]
This seems like 3 PRs to me:
1) Add timestamp to log (looks very committable if applied to all cases)
2) Don't overwrite log file (also very committable)
3) Redirect stdout to log (needs discussion)

Please separate #1 and #2 into separate PRs so they can be committed and leave this PR for the STDOUT topic.

File: bin/pgaudit_analyze (hunk 1 above)
Comment (amalek215) [reply bin/pgaudit_analyze:683]: ok, will do new PR shortly

File: bin/pgaudit_analyze (hunk 2 above)
Comment (amalek215) [reply bin/pgaudit_analyze:646]: the issue is when run as a systemd process without the --daemon flag all of the  `print` statements do not make it into the logfile
I will also make this its own PR

**Comment by dwsteele** [comment]
Closing since this PR was split into #18, #19, #20.

**Pull request closed without merging**